tests/chaos/utils.py

```diff
@@ -392,21 +393,42 @@ def rebooting_node(node, utility_pods):
     wait_for_node_status(node=node, status=True, wait_timeout=TIMEOUT_10MIN)
 
 
-def pod_deleting_process_recover(resource, namespace, pod_prefix):
+def pod_deleting_process_recover(resources, namespace, pod_prefix):
     """
-    This function will make sure that the pods for the affected deployment recover after the test.
+    Ensure that pods for affected resources (Deployment / DaemonSet) recover after the test.
+    'resources' can be a single resource or a list of resources.
     """
-    resource_objs = get_resources_by_name_prefix(
-        prefix=pod_prefix,
-        namespace=namespace,
-        api_resource_name=resource,
-    )
-    if not resource_objs:
-        raise ResourceNotFoundError(f"No resources found with prefix {pod_prefix} in namespace {namespace}")
+    if not isinstance(resources, (list, tuple)):
+        resources = [resources]
+
+    recovered_any = False
+
+    for resource in resources:
+        resource_objs = get_resources_by_name_prefix(
+            prefix=pod_prefix,
+            namespace=namespace,
+            api_resource_name=resource,
+        )
+        if not resource_objs:
+            LOGGER.debug(f"No {resource.__name__} found with prefix {pod_prefix} in namespace {namespace}")
+            continue
 
-    for resource_obj in resource_objs:
-        if resource_obj.kind == Deployment.kind:
-            resource_obj.wait_for_replicas()
+        for resource_obj in resource_objs:
+            recovered_any = True
+
+            if resource_obj.kind == Deployment.kind:
+                LOGGER.info(f"Waiting for Deployment {resource_obj.name} to recover replicas")
+                resource_obj.wait_for_replicas()
+
+            elif resource_obj.kind == DaemonSet.kind:
+                LOGGER.info(f"Waiting for DaemonSet {resource_obj.name} to recover")
+                resource_obj.wait_until_deployed()
+
+            else:
+                LOGGER.warning(f"Unsupported resource kind {resource_obj.kind} for recovery")
```

Comment:
should be checked at the begining of the function and raise if resource mismatch - always fail fast